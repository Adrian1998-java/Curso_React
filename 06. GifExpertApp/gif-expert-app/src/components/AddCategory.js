import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [Value, setValue] = useState('');

    const handleInputChange = ( e ) => {
        setValue( e.target.value );
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if( Value.trim().length > 2 ){
            setCategories( cats => [Value, ...cats]);
            setValue('');
        }
       
    }

  return (
    <>
    <form onSubmit={handleSubmit}>
        <p>{ Value }</p>
        <input 
            type='text'
            value={ Value }
            onChange={handleInputChange}
        />
    </form>
    </>
  )
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
