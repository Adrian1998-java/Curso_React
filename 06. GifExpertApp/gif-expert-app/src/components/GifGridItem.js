import React from 'react'

export const GifGridItem = ( {title, url} ) => {
  
    // console.log( img );

   return (
    <div className='card animate__animated animate__zoomIn'>
        <img src={url} alt={title}/>
        <p>{title}</p>
    </div>
  )
}

export default GifGridItem;
