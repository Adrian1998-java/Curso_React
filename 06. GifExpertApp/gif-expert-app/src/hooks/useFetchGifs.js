import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs= ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect( () => {
        
        getGifs( category )
            .then(img => {
                setTimeout(() => {
                    console.log(img)
                    setState({
                        data: img,
                        loading: false
                    })
                }, 1500);
            })    
        

    }, [category])

    return state;
}



