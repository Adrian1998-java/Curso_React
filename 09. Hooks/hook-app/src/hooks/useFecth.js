import { useEffect, useRef, useState } from "react"


export const useFecth = (url) => {
  
    const isMountet = useRef(true);
    const [state, setState] = useState( {data: null, loading: true, error: null} )

    useEffect(() => {
        
        return () => {
            isMountet.current = false;
        }

    }, [])
    

    useEffect(() => {

        setState({data: null, loading: true, error: null});

      fetch( url )
        .then( resp => resp.json() )
        .then( data => {

                if( isMountet.current ){
                    setState({
                        loading: false,
                        error: null,
                        data
                    })
                }
        } )
    }, [url])
    


    return state;
}
