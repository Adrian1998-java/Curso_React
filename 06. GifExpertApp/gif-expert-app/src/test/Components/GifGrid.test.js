import { shallow } from "enzyme"
import React from "react"
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Pruebas en GifGrid', () => { 
    
    const category = 'One punch';
    const wrapper = shallow( <GifGrid category={category}/> );

    test('Debe de mostrar el componente correctamente', () => { 
        
        useFetchGifs.mockReturn({
            data: [],
            loading: true
        });
        
        expect( wrapper ).toMatchSnapshot();
     });

     test('Debe de mostrar items cuando se cargan imagenes useFecth', () => { 
        const gifs = [{
            id:'ABS',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'CUalqwuier cosa',
        },
        {
            id:'ABFG',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier otra cosa',
        }]

        useFetchGifs.mockReturn({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category}/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false )
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
      })
 })