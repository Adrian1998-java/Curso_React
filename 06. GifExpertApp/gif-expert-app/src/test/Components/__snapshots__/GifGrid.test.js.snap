// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__slideInLeft"
  >
    Hola mundo
  </h3>
  <p
    className="card animate__animated animate__zoomIn"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
